src/utils/AddressUtils.ts

```diff
@@ -0,0 +1,18 @@
+import { providers } from "ethers";
+
+/**
+ * Checks if a contract is deployed at the given address
+ * @param address The ETH address to check
+ * @param provider A valid Ethers.js provider
+ * @returns A boolean indicating if a contract is deployed at the given address or not (true = contract, false = no contract)
+ */
+export async function isContractAddress(address: string, provider: providers.Provider): Promise<boolean> {
+  // A base case for if the address is null
+  if (!address) {
+    return false;
+  }
+  // Retrieve the code at the address
+  const code = await provider.getCode(address);
```

Comment:
This might be useful rather than rolling our own: https://docs.ethers.org/v5/api/utils/address/#utils-isAddress

Reply:
This function is used to determine if there's a contract deployed at an address. The `isAddress` is just a regex checker and does not determine if the address specified has a contract deployed to it. We'd ideally only want to relay a message to a contract that exposes `handleAcrossMessage` so this is an early signal that we have a malformed reqeust.

Reply:
hmmm great counterpart, ignore my suggestion then